### PR TITLE
530 edit podcast engagement

### LINKED ISSUE #530: Edit Podcast Engagement
After #505.  Build the Podcast -> Settings -> [Engagement Settings](https://www.figma.com/proto/Yewv83Q8btHcaOBWCtGgKs/PRX-Dovetail-with-Navigation?page-id=80%3A2947&node-id=2450%3A25654&viewport=798%2C-3032%2C0.25&scaling=min-zoom&starting-point-node-id=80%3A3420&hide-ui=1) form.

image

- [x] Setup the routes/tab. Use `podcast_engagement#index` and `/podcasts/123/engagement`
  - I could also be convinced this doesn't need its own controller, and should just use PodcastsController.  But we'll see.
- [ ] Use a bootstrap grid and other bootstrap-y things for the layout.
- [ ] Just add the existing subset of these fields right now:
  - Maaaaybe the homepage link, if we keep it on this tab
  - Donation form url
  - Micropayments wallet (just 1, not 2)
- [ ] Add the save button + last-saved

### PR DESCRIPTION
**Pull request by a-maci29**
Closes #530

### CODE REVIEW

File: app/views/podcast_engagement/show.html.erb [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+        </div>
+
+            <hr style="height:.1rem background-color:black">
+
+            <h3 aria-label="Micropayments"><%= t(".micropayments") %></h3>
+
+            <div class="form-floating input-group">
+                <%= form.text_field :payment_pointer %>
+                <label for="floatingInput"><%= t(".payment_pointer") %></label>
+                <span class="material-icons mx-auto justify-content-center col-1 border px-auto">link</span>
```
Comment (a-maci29) [inline app/views/podcast_engagement/show.html.erb:20]: Picking on how to align/add to the input seamlessly.

File: app/views/podcast_engagement/show.html.erb [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+        <div class="card-header-info">
+            <h5 class="card-title fw-bold">Status</h5>
+        </div>
 
-<%= render "form", podcast_engagement: @podcast %>
+        <div class="card-footer text-start">
+            <strong><%= t(".last_saved") %></strong>
+            <datetime><%= l(@podcast.updated_at) %></datetime>
+            <div class="btn-group">
+                <%= form.submit class: "btn btn-primary" %>
```
Comment (a-maci29) [inline app/views/podcast_engagement/show.html.erb:40]: Will adjust so the button and `Last Saved` text are side by side as in the [figma doc.](https://www.figma.com/proto/Yewv83Q8btHcaOBWCtGgKs/PRX-Dovetail-with-Navigation?page-id=80%3A2947&node-id=2450-25654&viewport=798%2C-3032%2C0.25&scaling=min-zoom&starting-point-node-id=80%3A3420&hide-ui=1)

Comment (cavis) [reply app/views/podcast_engagement/show.html.erb:40]: This seems pretty similar to the `views/podcasts/_form_status.html.erb`?  Which I fixed to be side-by-side in #629, if you want to dig through that code for an example.

**Review by cavis — CHANGES_REQUESTED** [review]
It begins with the comments!

File: app/controllers/podcast_engagement_controller.rb [hunk 3]
```diff
@@ -1,17 +1,16 @@
 class PodcastEngagementController < ApplicationController
   before_action :set_podcast
 
-  # GET /podcasts/1/engagement
-  def show
```
[the remark this answers is not in the document]
Comment (a-maci29) [reply app/controllers/podcast_engagement_controller.rb:5]: Does it need to have anything in it?

File: app/views/podcast_engagement/show.html.erb [hunk 4]
```diff
@@ -1,7 +1,47 @@
+<% content_for :title, @podcast.title.to_s %>
+
 <%= render "podcasts/tabs" %>
 
-<h1>Editing podcast engagement</h1>
+<%= form_with(model: @podcast, url: podcast_engagement_path, html: {autocomplete: "off"}, data: {controller: "unsaved", unsaved_changed_value: @podcast.changed?, unsaved_confirm_value: t(".confirm")}) do |form| %>
```
[the remark this answers is not in the document]
Comment (a-maci29) [reply app/views/podcast_engagement/show.html.erb:5]: Actually I have changed my mind, but to my credit, different question: What should I be seeing when testing the `confirm` translation and changing tabs?

File: app/views/podcast_engagement/show.html.erb [hunk 5]
```diff
@@ -4,47 +4,38 @@
...
-                    <div class="input-group">
-                        <div class="form-floating">
-                            <%= form.text_field :donation_url %>
-                            <%= form.label :donation_url %>
-                        </div>
-
-                        <div class="input-group-text"> <span class="material-icons">link</span> </div>
+    <div class="row my-4">
+        <div class="col-6">
+            <div class="col-12 mb-4">
```
Comment (a-maci29) [inline app/views/podcast_engagement/show.html.erb]: May be a good @brandonhundt question; tried to follow the pattern of other forms, but the `col-6` kept the form/input fields sized correctly, trying to wrap it in the `col-12` instead doesn't.
Wouldn't the `col-12` denote the entire width of the wrapping container, and `col-6` use half of that?

Comment (brandonhundt) [reply app/views/podcast_engagement/show.html.erb]: Try this starting at line 7...

```
<div class="col-lg-8">
  <div class="row">
    <div class="col-12 mb-4">
    
```

then the side bar would be:

```
<div class="col-lg-4">
  <div class="row">
```

File: app/views/podcast_engagement/show.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (cavis) [reply app/views/podcast_engagement/show.html.erb:4]: It should pop up a modal (not an HTML one, but a built-in browser style OK/cancel one) ... with the text you added to en.yml in it.

File: app/controllers/podcast_engagement_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (cavis) [reply app/controllers/podcast_engagement_controller.rb:5]: Nope - just the code that's there in `main` right now.

File: app/controllers/podcast_engagement_controller.rb [hunk 6]
```diff
@@ -15,6 +15,9 @@ def update
     end
   end
 
+  def show
+  end
```
[the remark this answers is not in the document]
Comment (cavis) [reply app/controllers/podcast_engagement_controller.rb:19]: Nothing - but I might move it back to where it was, so the diff is cleaner here.

**Pull request merged**